openhcl/virt_mshv_vtl/src/processor/mshv/x64.rs

```diff
@@ -1293,7 +1297,9 @@ impl UhProcessor<'_, HypervisorBackedX86> {
     }
 }
 
-impl<T: CpuIo> EmulatorSupport for UhEmulationState<'_, '_, T, HypervisorBackedX86> {
+impl<T: CpuIo> EmulatorSupport
+    for UhEmulationState<'_, '_, T, HypervisorBackedX86, x86emu::CpuState>
```

Comment:
Take a look at #478 for how I hang this type off of the Backing as an associated type, instead of needing a new generic parameter. That feels cleaner to me, since I don't think we'll ever have a case where one backing needs to support different state types, right?

Reply:
Yes!!! Your reference was very helpful here, I also was not a fan of the new generic parameter. I pushed an alternative with an associated type.